internal/pkg/gateway/commit/transactionstatus.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/common/ledger/blockledger"
+	"github.com/pkg/errors"
+)
+
+// TransactionStatus is sent to the returned channel when received from the block iterator. The returned channel will
+// be closed after the status is sent, or if an error occurs that prevents the transaction status from being obtained.
+// The receiver should always check the the closed status of the channel on reads.
+func TransactionStatus(blockIterator blockledger.Iterator, transactionID string, quit <-chan int) <-chan peer.TxValidationCode {
```

Comment:
Since this is not wired, let's go ahead and add the `ctx context.Context` parameter and check for the context ending in the select?  Two modified lines in the function, and a new test, so not too intrusive.

Reply:
I've pushed up a change with what I think you're asking for. The only difference is the code now needs to pull the done channel out of the context argument rather than having the done channel passed in as an argument. Since the code doesn't require anything else from the context, I'm not convinced it's an improvement